src/content/structured/get-started/testing-with-react-testing-library.mdx

```diff
@@ -0,0 +1,231 @@
+---
+path: "/get-started/install-components/testing-with-react-testing-library"
+
+navPriority: 8
+
+date: "2023-01-20"
+
+title: "Testing with React Testing Library"
+
+subTitle: "How to test components with React Testing Library."
+
+contribute: "https://github.com/mi6/ic-design-system/tree/main/src/content/structured/get-started/testing-with-react-testing-library.mdx"
+---
+
+## About React Testing Library
+
+[React Testing Library (RTL)](https://testing-library.com/docs/react-testing-library/intro) provides a set of utility functions which interrogate the DOM tree in order to build maintainable React tests. The recommended approach with React Testing Library is to test React components based on how the users view the components rather than the implementation of the component. For example, a user would view a button by the label 'Add', so the library provides a method called `getByText()` to facilitate that.
+
+## Shadow DOM Testing Library
+
+However, RTL does not provide utility functions which go beyond the `shadowDOM`. As `@ukic/react` are React wrapped web components, which use the shadowDOM, it is not possible to exclusively use RTL to create React tests. Fortunately, with the addition of [Shadow DOM Testing Library](https://github.com/KonnorRogers/shadow-dom-testing-library), it is possible to select elements in the shadowDOM and interact with them by using methods provided by RTL. Shadow DOM Testing Library provides functions with 'Shadow' prefixed prior to the query type (e.g. `getByShadowText()`).
```

Comment:
```suggestion
However, RTL does not provide utility functions which go beyond the `shadowDOM`. As `@ukic/react` are React-wrapped web components, which use the shadowDOM, it is not possible to exclusively use RTL to create React tests. Fortunately, with the addition of [Shadow DOM Testing Library](https://github.com/KonnorRogers/shadow-dom-testing-library), it is possible to select elements in the shadowDOM and interact with them by using methods provided by RTL. Shadow DOM Testing Library provides functions with 'Shadow' prefixed prior to the query type (e.g. `getByShadowText()`).
```

Reply:
I don't think a hyphen is required here for 'React-wrapped'. It would make more sense for it to have no hyphen as it could potentially confuse the reader as it is not code.

Reply:
@GCHQ-Developer-465 Hey, a hyphen is required because it's a [compound adjective](https://www.grammarly.com/blog/compound-adjectives/). We're using "React-wrapped" to describe the web components in this context